### PR TITLE
upgrade dependencies and fix tests that broke when upgrading chai

### PR DESCRIPTION
**Pull request by saponifi3d**
This upgrades handlebars, underscore, peer dependency on 1.0.0 for rendr (although it's probably not need, i haven't tested this stuff against older versions).

the test changes were needed because of the update to chai.

### CODE REVIEW

File: test/shared/helpers/forEach.test.js [hunk 1]
```diff
@@ -103,11 +103,11 @@ describe('forEach', function () {
       var thisCall = spy.getCall(0),
           args = thisCall.args[0];
 
-      expect(args).to.deep.property('key', 0);
```
Comment (jmerrifield) [inline test/shared/helpers/forEach.test.js]: You probably just needed to change these to `.to.have.property('key', value)`. These matchers will give you a much better failure message if you can figure out how to make them work.

Comment (saponifi3d) [inline test/shared/helpers/forEach.test.js]: :+1: seems legit - didn't think about that one.  updated it and the rest of the cases that i *can* there are some things that i'm test for deep equal that can't be compared this way :(

**Comment by roncli** [comment]
FYI handlebars is now on 3.0.0.  Unfortunately, this doesn't appear to be as easy as just upgrading the dependency this time around.

**Comment by saponifi3d** [comment]
@roncli - agreed. We've been looking at a few things to each app to define which version of handlebars you want to use so we don't have to manage it here.

This is the PR for it: https://github.com/rendrjs/rendr-handlebars/pull/52

**Pull request merged**